src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeTypeBuilder.cs

```diff
@@ -111,13 +111,12 @@ private sealed class CustAttr
             private readonly byte[]? m_binaryAttribute;
             private readonly CustomAttributeBuilder? m_customBuilder;
 
-            public CustAttr(ConstructorInfo con, byte[] binaryAttribute)
+            public CustAttr(ConstructorInfo con, ReadOnlySpan<byte> binaryAttribute)
             {
                 ArgumentNullException.ThrowIfNull(con);
-                ArgumentNullException.ThrowIfNull(binaryAttribute);
 
                 m_con = con;
-                m_binaryAttribute = binaryAttribute;
+                m_binaryAttribute = binaryAttribute.ToArray();
```

Comment:
This is introducing an extra array copy. Do you remember whether this regression was considered during the API review discussion that proposed changing the argument type to ReadOnlySpan?

Reply:
> Do you remember whether this regression was considered during the API review discussion that proposed changing the argument type to ReadOnlySpan?

No, I don't think so, there were not much discussion about this change, quickly agreed for this update.  Would you suggest to keep the `SetCustomAttributeCore(ConstructorInfo con, byte[] binaryAttribute)` for now?

Reply:
I think that the extra copy is fine, it is just something that I have noticed and that I did not expect. I do not have a strong opinion about it. What do you think?

Reply:
I think this path would be used rarely (attributes for generic type parameter) so probably fine, but for the new assembly builder implementation we are creating this copy for each attribute: https://github.com/dotnet/runtime/blob/02536ba141ade74355dcfef8a8d52a2b7aaa2f54/src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/CustomAttributeWrapper.cs#L16
I think I can workaround that to keep the BlobHandle of the `ReadOnlySpan<byte>` instead. But that needs https://github.com/dotnet/runtime/pull/81059 merged and a public API for `ReadOnlySpan<byte>` overload is added

Reply:
Added TODO for using `BlobHandle` instead of `byte[]`. 

There is similar allocation added in mono: https://github.com/dotnet/runtime/blob/a7ed5ce7ef03dacb7662e58bead008d118f7125f/src/mono/System.Private.CoreLib/src/System/Reflection/Emit/CustomAttributeBuilder.Mono.cs#L85 It is OK in case the `SetCustomAttribute(ConstructorInfo con, byte[] binaryAttribute)` overload used directly as previously it was also cloning the byte array https://github.com/dotnet/runtime/blob/2087ceb0c70f0c13e79fbbdec90efd242e1f2cd4/src/mono/System.Private.CoreLib/src/System/Reflection/Emit/CustomAttributeBuilder.Mono.cs#L86 but as you know if the `SetCustomAttribute(CustomAttributeBuilder customBuilder)` overload is used it will call `SetCustomAttributeCore(customBuilder.Ctor, customBuilder.Data)` and going to create another copy ...